tesseract_core/sdk/docker_client.py

```diff
@@ -35,21 +35,26 @@ def _get_executable(program: Literal["docker", "docker-compose"]) -> tuple[str,
 class Image:
     """Image class to wrap Docker image details."""
 
-    id: str
-    short_id: str
+    id: str | None
+    short_id: str | None
     tags: list[str] | None
     attrs: dict
 
     @classmethod
     def from_dict(cls, json_dict: dict) -> "Image":
         """Create an Image object from a json dictionary."""
         image_id = json_dict.get("Id", None)
-        if image_id and not image_id.startswith("sha256:"):
-            # Some container engines (e.g., Podman) do not prefix the ID with sha256:
-            image_id = f"sha256:{image_id}"
+        short_id = None
+        if image_id:
+            if image_id.startswith("sha256:"):
+                short_id = image_id[:19]
+            else:
+                # Some container engines (e.g., Podman) do not prefix IDs with sha256
+                short_id = image_id[:12]
```

Comment:
Any reason we don't go even shorter?

Reply:
This is what Docker expects. You can reference images via full ID, short ID, or tag. Short ID must be 12 characters long or you'll get an error.